packages/core/src/execution-engine/node-execution-context/node-execution-context.ts

```diff
@@ -178,6 +180,22 @@ export abstract class NodeExecutionContext implements Omit<FunctionsBase, 'getCr
 		return this.instanceSettings.instanceId;
 	}
 
+	protected getExecutionWaitingToken() {
+		const token = crypto
+			.createHmac('sha256', this.instanceSettings.encryptionKey)
```

Comment:
We should **not** use the instance encryption key directly for this as it’s not safe. Instead we should let it be configurable and by default derive a nee key from the encryption key. See eg the binary file signing key https://github.com/n8n-io/n8n/blob/f30cc7b6cfba6998091f31fcd3012a971b3a2bb8/packages/core/src/binary-data/binary-data.config.ts#L30